src/hotspot/share/prims/jvmtiEnv.cpp

```diff
@@ -1074,18 +1074,10 @@ JvmtiEnv::SuspendAllVirtualThreads(jint except_count, const jthread* except_list
 // thread - NOT protected by ThreadsListHandle and NOT pre-checked
 jvmtiError
 JvmtiEnv::ResumeThread(jthread thread) {
-  JvmtiVTMSTransitionDisabler disabler(true);
-  JavaThread* current = JavaThread::current();
-  ThreadsListHandle tlh(current);
-
-  JavaThread* java_thread = nullptr;
-  oop thread_oop = nullptr;
-  jvmtiError err = get_threadOop_and_JavaThread(tlh.list(), thread, current, &java_thread, &thread_oop);
-  if (err != JVMTI_ERROR_NONE) {
-    return err;
-  }
-  err = resume_thread(thread_oop, java_thread, /* single_resume */ true);
-  return err;
+  // resume thread with handshake
+  ResumeThreadClosure op(/* single_resume */ true);
```

Comment:
Could you please explain how thread is protected from racing with mounting<->unmounting operations  with resume_thread operations? 
It might be unlikely happens for suspended threads, but for alive threads the results are not defined.

Reply:
Thank you for the question. The `JvmtiHanshake::execute()` has a `JvmtiVTMSTransitionDisabler` installed:
```
JvmtiHandshake::execute(JvmtiUnitedHandshakeClosure* hs_cl, jthread target) {
  JavaThread* current = JavaThread::current();
  HandleMark hm(current);
  JvmtiVTMSTransitionDisabler disabler(target);            <= !!!!!!!
  . . .
```